go-controller/pkg/util/node_annotations.go

```diff
@@ -1330,12 +1356,15 @@ func parseNetworkMapAnnotation(nodeAnnotations map[string]string, annotationName
 func ParseNetworkIDAnnotation(node *corev1.Node, netName string) (int, error) {
 	networkIDsMap, err := parseNetworkMapAnnotation(node.Annotations, ovnNetworkIDs)
 	if err != nil {
-		return InvalidID, err
+		if IsAnnotationNotSetError(err) {
+			return types.NoNetworkID, err
+		}
+		return types.InvalidID, err
 	}
 
 	networkID, ok := networkIDsMap[netName]
 	if !ok {
-		return InvalidID, newAnnotationNotSetError("node %q has no %q annotation for network %s", node.Name, ovnNetworkIDs, netName)
+		return types.NoNetworkID, newAnnotationNotSetError("node %q has no %q annotation for network %s", node.Name, ovnNetworkIDs, netName)
```

Comment:
I find this function returning different errors confusing, not sure why anyone would want to distinguish empty annotation that will return InvalidID vs unset annotation that will return NoNetworkID. It also complicates error handling on the caller side (e.g. currently in `syncNodeNetworkAnnotations` you only check for NoNetworkID, but I think InvalidID should have the same effect ).